Projects/Contracts/Sources/Contracts.swift

```diff
@@ -36,36 +36,58 @@ public indirect enum ContractFilter {
 
 public struct Contracts {
     let filter: ContractFilter
-    let state = ContractsState()
-
-    public static var openFreeTextChatHandler: (_ viewController: UIViewController) -> Void = { _ in }
     public init(filter: ContractFilter = .active(ifEmpty: .terminated(ifEmpty: .none))) { self.filter = filter }
 }
 
 public enum ContractsResult {
     case movingFlow
+    case openFreeTextChat
 }
 
 extension Contracts: Presentable {
     public func materialize() -> (UIViewController, Signal<ContractsResult>) {
         let viewController = UIViewController()
 
+        let store: ContractStore = get()
+
+        let bag = DisposeBag()
+
         if filter.displaysActiveContracts {
             viewController.title = L10n.InsurancesTab.title
             viewController.installChatButton()
         }
 
-        let bag = DisposeBag()
-
         bag += viewController.install(
-            ContractTable(presentingViewController: viewController, filter: filter, state: state)
+            ContractTable(presentingViewController: viewController, filter: filter)
         )
 
+        // Initial fetch
+        store.send(.fetchContracts)
```

Comment:
Would it be a bad idea to add an effect on just `.fetch` that dispatches the three events?